### PR TITLE
docs: reset Forge Kernel authority plan

### PR DESCRIPTION
**Pull request by harshanandak**
## Problem
The v3 plan still had Beads/Dolt framed as the runtime authority in parts of the release plan, locked decisions, and workflow assembly docs. That conflicted with the newer Forge Kernel direction: Forge should own issue, workflow, run, provider, and projection state while Beads becomes import/export compatibility.

## Root Cause
The previous workflow assembly/control-plane plan evolved around Beads as the durable tracker. After the architecture shifted to a Forge Kernel with local SQLite authority and later Cloudflare team authority, the canonical docs and release train needed to be reset together so future implementation does not drift back into Beads-as-authority decisions.

## Fix
This PR adds the Forge Kernel authority control-plane plan, storage model, and decision drift guards. It updates the release plan, locked decisions, docs index, capability-pack docs, harness parity docs, and changelog to use Kernel authority, provider capability contracts, explicit storage boundaries, and `backstop_only` terminology instead of fallback/Beads authority wording.

## Value
This gives future 0.0.20+ implementation a clear source of truth: local mode uses a SQLite WAL broker, team mode moves to Cloudflare authority, Beads is import/export only, and external skills/MCPs/agents/commands/hooks/scripts/extensions are governed as provider capabilities through Forge Kernel configuration and evaluator evidence.

## Beads
Closes forge-2agy.8

<details>
<summary>Implementation Details</summary>

### Test Coverage
- Tests: `bun run check` passed; validation test shard reported 810 passing, 0 failing before the changelog commit and 802 passing / 22 skipped / 0 failing in the post-changelog run.
- Pre-push targeted tests: 27 passing, 0 failing.
- Scenarios covered: docs index/reference consistency, workflow stage docs checks, command sync drift checks, edge-case security tests, package skill tests after workspace dependencies were installed.

### Security Review
- OWASP Top 10: docs-only architecture change; no runtime input handling, auth, storage code, or network path added.
- Automated scan: `bun audit` reported one moderate transitive `qs` advisory through `@stryker-mutator/core`; `scripts/validate.js` treats moderate/low findings as non-blocking and blocks only high/critical findings.

### Design Doc
See: `docs/work/2026-04-28-skeleton-pivot/forge-kernel-authority-control-plane.md`

### Key Decisions
- Forge Kernel is the issue/workflow/run authority; Beads is import/export only.
- Local single-user/multi-worktree mode uses a SQLite WAL broker keyed by git common-dir.
- Team mode uses Cloudflare Worker API, Durable Object write authority, D1 read model, Queues projections, and optional R2 evidence archive.
- Skills, MCPs, agents, commands, hooks, scripts, docs/context packs, and external extensions are provider capabilities controlled by Kernel config.
- Decision drift guards now make Beads-as-authority, silent fallback, and direct generated/harness authority writes explicit forbidden patterns.

### Documentation Updated
- `CHANGELOG.md`
- `docs/INDEX.md`
- `docs/reference/AGENT_SKILL_PARITY.md`
- `docs/reference/DECISION_DRIFT_GUARDS.md`
- `docs/reference/FORGE_KERNEL_STORAGE_MODEL.md`
- `docs/work/2026-04-28-skeleton-pivot/forge-kernel-authority-control-plane.md`
- `docs/work/2026-04-28-skeleton-pivot/locked-decisions.md`
- `docs/work/2026-04-28-skeleton-pivot/release-plan.md`
- `docs/work/2026-04-28-skeleton-pivot/week-3-runtime-capability-packs.md`

### Validation
- [x] Type check passing
- [x] Lint passing (0 errors, 0 warnings)
- [x] All tests passing
- [x] Security review completed

</details>

---

### Self-Review Checklist

- [x] I reviewed the full diff on GitHub
- [x] No debug code (console.log, commented code, temporary changes)
- [x] ESLint passes with zero warnings (`bun run lint`)
- [x] All tests pass locally (`bun run check`)
- [x] No hardcoded secrets or API keys
- [x] No breaking changes (or documented in migration guide)
- [x] TDD compliance: Documentation-only architecture plan; no source files changed

## Summary by CodeRabbit

* **Documentation**
  * Added new Forge Kernel planning references covering authority, storage, drift safeguards, and release evaluation.
  * Expanded the main index and team workflow docs with new architecture links and updated guidance.
  * Refreshed release planning, locked decisions, and strategy docs to align with the Kernel-first model and current roadmap.
  * Updated agent capability and runtime docs with terminology and formatting refinements.
  * Added a changelog entry summarizing the authority plan reset.

[Review Change Stack](https://app.coderabbit.ai/change-stack/harshanandak/forge/pull/191?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: docs/work/2026-04-28-skeleton-pivot/locked-decisions.md [hunk 1]
```diff
@@ -560,11 +560,29 @@ ACTIVE.
 
 ---
 
+## D44 - Forge Kernel replaces Beads/Dolt as issue authority; Beads becomes import/export adapter
+
+**Decision**: Forge adopts a native issue authority model: Forge Kernel API, local SQLite WAL broker for solo multi-worktree coordination, and optional Cloudflare team authority for multi-user mode. Beads is no longer the target runtime authority. Beads remains a migration source and optional projection/export adapter. GitHub and Linear are server-side projections, not local runtime authorities. ACTIVE - supersedes Beads-only portions of D21, D22, D23, D30, D31, and D36.
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply docs/work/2026-04-28-skeleton-pivot/locked-decisions.md:569]: Fixed in b0910a7: AGENTS.md now points current authority work to Forge Kernel/D44, and v3-redesign-strategy explicitly supersedes Beads/Dolt default-substrate language with the Forge Kernel authority control plane.

File: docs/work/2026-04-28-skeleton-pivot/locked-decisions.md [hunk 2]
```diff
@@ -236,7 +236,7 @@ When a doc disagrees with this file, this file wins until a successor decisions
...
+**Decision**: Forge ships `.forge/protected-paths.yaml` defining seven categories of protected paths: `forge_core` (checksum-verified), `user_protocol` (CLI-only mods), `generated_artifacts` (CI-blocked hand-edits), `append_only_logs` (runtime-only writes), `secrets` (already covered by rail #2), `beads_state` (bd CLI only), `immutable` (`.git`, etc). Enforcement layers: per-harness PreToolUse hooks (Claude Code + Codex CLI), Cursor file-watcher backstop, pre-commit lefthook entry, session-start checksum verification, CI lint job. Refuse-with-hint UX guides agents toward the proper CLI commands. **Total L1 rail count stays at 5** — this expands rail #5 (schema + integrity) scope; it does not add a new rail.
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply docs/work/2026-04-28-skeleton-pivot/locked-decisions.md]: Fixed in b0910a7: changed the protected-path category list from etc to etc. in locked-decisions.md.

File: docs/work/2026-04-28-skeleton-pivot/locked-decisions.md [hunk 3]
```diff
@@ -560,11 +560,29 @@ ACTIVE.
 
 ---
 
+## D44 - Forge Kernel replaces Beads/Dolt as issue authority; Beads becomes import/export adapter
+
+**Decision**: Forge adopts a native issue authority model: Forge Kernel API, local SQLite WAL broker for solo multi-worktree coordination, and optional Cloudflare team authority for multi-user mode. Beads is no longer the target runtime authority. Beads remains a migration source and optional projection/export adapter. GitHub and Linear are server-side projections, not local runtime authorities. ACTIVE - supersedes Beads-only portions of D21, D22, D23, D30, D31, and D36.
+
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply docs/work/2026-04-28-skeleton-pivot/locked-decisions.md:570]: Fixed in b0910a7: locked-decisions.md top metadata now tracks D44 and the 2026-05-29 authority reset.

File: docs/work/2026-04-28-skeleton-pivot/release-plan.md [hunk 4]
```diff
@@ -19,7 +19,9 @@ Do not describe active package plans with the old major-version labels.
...
+After `0.0.18`, keep the same release discipline but shift the product lens from "board/dashboard" to "Forge Kernel authority control plane." Boards remain views over the runtime ledger. Forge owns the issue graph, local broker, protected state surfaces, memory projections, workflow assembly, and adapter contracts; Beads, GitHub, Linear, Claude, Cursor, Codex, and other agents are projections or adapters over that control plane.
+
+**Authority reset (2026-05-29)**: Forge is internal-only, so the post-0.0.18 plan no longer treats Beads/Dolt compatibility as a public-user constraint. The active architecture is [Forge Kernel authority control plane](./forge-kernel-authority-control-plane.md): local SQLite WAL broker first, Cloudflare team authority second, Beads as import/export projection only. The evaluator loop in that plan scores the final architecture **100/100** across architecture, security/privacy, user perspective, UX, edge cases, implementation simplicity, scalability, and plan alignment.
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply docs/work/2026-04-28-skeleton-pivot/release-plan.md:26]: Fixed in b0910a7: release-plan.md now has Last updated: 2026-05-29 for the Forge Kernel authority reset.
